ui/src/app/job-list/table/table.component.ts

```diff
@@ -101,11 +118,24 @@ export class JobsTableComponent implements OnInit {
     return "https://www.gstatic.com/images/icons/material/system/1x/arrow_drop_down_grey700_24dp.png"
   }
 
-  getJobLabel(job: QueryJobsResult, label: string): string {
-    if (job.labels && job.labels[label]) {
-      return job.labels[label];
+  isStatusField(df: DisplayField): boolean {
+    return df.field == "status";
+  }
+
+  getFieldValue(job: QueryJobsResult, df: DisplayField): any {
+    // Handle nested fields, separated by '.', i.e. extensions.userId
+    let fields = df.field.split(".");
+    let value = job;
+    for (let field of fields) {
+      value = value[field];
```

Comment:
Slightly worried about a tough-to-debug error here as a result of returning a malformed field path from the capabilities endpoint. Perhaps exit early if value is null.

Reply:
Good point, done.